File: benchmarks/word_count.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
d = Hash.new(0)
i = 0
while i < 1000000
    w = "word#{i % 1000}"
    d[w] += 1
    i += 1
end
total = 0
d.each_value do |v|
    total += v
end
puts (total + d.size)
puts "bench_ns=#{Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond) - t0}"

File: benchmarks/word_count.rs
use std::time::Instant;
use std::collections::HashMap;

fn main() {
    let t0 = Instant::now();
    let mut d: HashMap<String, i32> = HashMap::new();
    for i in 0..1000000 {
        let w = format!("word{}", i % 1000);
        *d.entry(w).or_insert(0) += 1;
    }
    let mut total = 0;
    for v in d.values() {
        total += v;
    }
    println!("{}", total + d.len() as i32);
    println!("bench_ns={}", t0.elapsed().as_nanos());
}
